checker/src/main/java/org/checkerframework/checker/confidential/ConfidentialTransfer.java

```diff
@@ -0,0 +1,128 @@
+package org.checkerframework.checker.confidential;
+
+import javax.lang.model.element.AnnotationMirror;
+import org.checkerframework.dataflow.analysis.TransferInput;
+import org.checkerframework.dataflow.analysis.TransferResult;
+import org.checkerframework.dataflow.cfg.node.Node;
+import org.checkerframework.dataflow.cfg.node.StringConcatenateNode;
+import org.checkerframework.framework.flow.CFAbstractAnalysis;
+import org.checkerframework.framework.flow.CFStore;
+import org.checkerframework.framework.flow.CFTransfer;
+import org.checkerframework.framework.flow.CFValue;
+import org.checkerframework.framework.type.QualifierHierarchy;
+import org.checkerframework.javacutil.AnnotationUtils;
+
+/** The transfer class for the Confidential Checker. */
+public class ConfidentialTransfer extends CFTransfer {
+
+  /** The Confidential type factory. */
+  protected final ConfidentialAnnotatedTypeFactory atypeFactory;
+
+  /** The Confidential qualifier hierarchy. */
+  protected final QualifierHierarchy qualHierarchy;
+
+  /**
+   * Create a new ConfidentialTransfer.
+   *
+   * @param analysis the corresponding analysis
+   */
+  public ConfidentialTransfer(CFAbstractAnalysis<CFValue, CFStore, CFTransfer> analysis) {
+    super(analysis);
+    atypeFactory = (ConfidentialAnnotatedTypeFactory) analysis.getTypeFactory();
+    qualHierarchy = atypeFactory.getQualifierHierarchy();
+  }
+
+  /**
+   * Enforces Confidential String concatenation rules:
+   *
+   * <ul>
+   *   <li>(Confidential + NonConfidential) returns Confidential (commutatively);
```

Comment:
This is the same comment text as in the TreeAnnotator.  Can you please document the relationship between the two?  What does each one do, and why are both necessary?